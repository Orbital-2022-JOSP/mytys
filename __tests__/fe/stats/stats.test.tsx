import { render } from '@testing-library/react';
import { Stats } from '../../../components/Stats/Stats';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {};
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated' }
        }),
    };
});

describe("The Stats Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<Stats />);

            expect(container).toMatchSnapshot()
        })
})

