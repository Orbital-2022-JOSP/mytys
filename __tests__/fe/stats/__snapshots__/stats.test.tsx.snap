// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Stats Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-20 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl md:px-24 lg:px-8 lg:py-25"
  >
    <div
      class="grid grid-cols-2 gap-8 md:grid-cols-4"
    >
      <div
        class="text-center md:border-r"
      >
        <h6
          class="text-4xl font-bold lg:text-5xl xl:text-6xl hover:text-light-blue-500"
        >
          10K
        </h6>
        <p
          class="text-sm font-medium tracking-widest text-gray-500 uppercase lg:text-base"
        >
          Students
        </p>
      </div>
      <div
        class="text-center md:border-r"
      >
        <h6
          class="text-4xl font-bold lg:text-5xl xl:text-6xl hover:text-light-blue-500"
        >
          200
        </h6>
        <p
          class="text-sm font-medium tracking-widest text-gray-500 uppercase lg:text-base"
        >
          Onboarded teachers
        </p>
      </div>
      <div
        class="text-center md:border-r"
      >
        <h6
          class="text-4xl font-bold lg:text-5xl xl:text-6xl hover:text-light-blue-500"
        >
          97.6%
        </h6>
        <p
          class="text-sm font-medium tracking-widest text-gray-500 uppercase lg:text-base"
        >
          Improvement in results
        </p>
      </div>
      <div
        class="text-center"
      >
        <h6
          class="text-4xl font-bold lg:text-5xl xl:text-6xl hover:text-light-blue-500"
        >
          95.9%
        </h6>
        <p
          class="text-sm font-medium tracking-widest text-gray-500 uppercase lg:text-base "
        >
          Student satisfaction
        </p>
      </div>
    </div>
  </div>
</div>
`;
